tests/pick-test.c

```diff
@@ -0,0 +1,220 @@
+#include <sys/ioctl.h>
+#include <sys/select.h>
+#include <sys/wait.h>
+
+#include <err.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+static char	**pickargv;
+static int	  gotsig;
+
+static char *
+parsekeys(const char *s)
+{
+	char	*buf;
+	size_t	 len = 0;
+	size_t	 size = 16;
+	int	 c;
+
+	if ((buf = malloc(size)) == NULL)
+		err(1, NULL);
+
+	for (; (c = *s) != '\0'; s++) {
+		if (c == '\\') {
+			switch (*++s) {
+			case 'n':
+				c = '\n';
+				break;
+			default:
+				c = *s;
+			}
+		} else if (c == ' ') {
+			continue;
+		}
+		buf[len++] = c;
+
+		if (size <= len) {
+			if ((buf = realloc(buf, 2*size)) == NULL)
+				err(1, NULL);
+			size *= 2;
```

Comment:
Totally doesn't matter for this test script, because we control all the input, but this could technically lead to overflow. Not saying you should do anything about this though.